lib/stream.js

```diff
@@ -39,14 +45,21 @@ Stream.PassThrough = require('_stream_passthrough');
 Stream.pipeline = pipeline;
 Stream.finished = eos;
 
-Stream.promises = promises;
+ObjectDefineProperty(Stream, 'promises', {
+  configurable: true,
+  enumerable: true,
+  get() {
+    if (!promises) promises = require('internal/streams/promises');
+    return promises;
+  }
+});
```

Comment:
@mcollina: If we lazy load here I see no point in lazy loading inside promises as well?

Reply:
I think they are all needed, because users may required via `steams/promises` instead of `steams.promises`.

Reply:
If you require stream/promises then you probably intend to use them, thus lazy loading would not be required.

Reply:
But I don't use all the functions in `promises.js`, I might just use the `pipeline`.